examples/iterable-objects/iterable_objects.bal

```diff
@@ -1,36 +1,89 @@
 import ballerina/io;
 
-// An object that is a subtype of `Iterator<int>`.
+// A class that is a subtype of `Iterator<int,()>`.
 class ArrayIterator {
     private int[] integers = [0, 1, 1, 2, 3, 5, 8, 13, 21, 34];
     private int cursor = -1;
 
     // `next` method which generates the sequence of values of type `int`.
-    public function next() returns record {|int value;|}? {
+    public isolated function next() returns record {| int value; |}? {
         self.cursor += 1;
-        if (self.cursor < self.integers.length()) {
-            record {|int value;|} nextVal = {value: self.integers[self.cursor]};
+        if self.cursor < self.integers.length() {
+            record {| int value; |} nextVal = 
+                {value: self.integers[self.cursor]};
             return nextVal;
+        } else if self.cursor > self.integers.length() {
+            panic error("Index out of range");
         }
         return ();
     }
 }
 
-//  An object that is a subtype of `Iterable<int>`.
+//  A class that is a subtype of `Iterable<int,()>`.
 class IteratorGenerator {
-
-    // The `__iterator()` method should return a new `Iterator<T>`.
-    public function __iterator() returns object {
-            public function next() returns record {|int value;|}?;} {
+    *object:Iterable;
+    // The `iterator()` method should return a new `Iterator<T,()>`.
```

Comment:
Since we are being specific here shouldn't we say

```suggestion
    // The `iterator()` method should return a subtype of `Iterator<int, ()>`.
```